config/config.go

```diff
@@ -268,6 +276,25 @@ func mergeConfigWithIncludedConfig(config *TerragruntConfig, includedConfig *Ter
 	return includedConfig, nil
 }
 
+// Merge the extra arguments prioritizing those defined in the childExtraArgs
+func mergeExtraArgs(terragruntOptions *options.TerragruntOptions, childExtraArgs []TerraformExtraArguments, parentExtraArgs *[]TerraformExtraArguments) {
+	result := *parentExtraArgs
+addExtra:
+	for _, child := range childExtraArgs {
+		for i, parent := range result {
+			if parent.Name == child.Name {
+				terragruntOptions.Logger.Printf("extra_arguments '%v' from child overriding parent", child.Name)
+				// For extra args, we want to keep the values specified in the child and put them after
+				// the parent ones, so if we encounter a duplicate, we just overwrite it.
+				result[i] = child
+				continue addExtra
```

Comment:
Yes, moving it into a separate function is the idea, just to improve readability.

Go, unfortunately, does not include a built-in `contains` method for arrays. You just implement it yourself using a loop. And since there are no generics either, you have to implement it separately for each type... The language definitely has some weaknesses :)